File: app/SweeperSD/app/src/main/java/com/example/joseph/sweepersd/SweeperSDApplication.java
package com.example.joseph.sweepersd;

import android.app.Application;

public class SweeperSDApplication extends Application {
    public static volatile boolean needsCoarsePermission;
    public static volatile boolean needsFinePermission;


    @Override
    public void onTrimMemory(int level) {
        super.onTrimMemory(level);
        if (level > Application.TRIM_MEMORY_BACKGROUND) {
            trimMemory();
        }
    }

    private void trimMemory() {
        /*WatchZoneRepository.getInstance(this).delete();
        WatchZoneModelUpdater.getInstance(this).delete();
        WatchZoneModelRepository.getInstance(this).delete();
        LimitRepository.getInstance(this).delete();*/
    }
}

File: app/SweeperSD/app/src/main/java/com/example/joseph/sweepersd/SweeperSDApplication.kt
package com.example.joseph.sweepersd

import android.app.Application
import com.example.joseph.sweepersd.bluetooth.BluetoothRecordRepo

@Suppress("unused")
class SweeperSDApplication : Application() {
    override fun onCreate() {
        super.onCreate()

        BluetoothRecordRepo.applicationContext = this
    }
}